plugin-api/src/main/java/org/sonarsource/sonarlint/plugin/api/SonarLintContext.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * SonarLint Plugin API
+ * Copyright (C) 2016-2021 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.sonarlint.plugin.api;
+
+import org.sonar.api.Plugin;
+
+/**
+ * This interface is implemented by the Plugin.Context instance provided to plugins in {link {@link org.sonar.api.Plugin#define(Plugin.Context)}}
+ *
+ * The specific instance can be retrieved in the plugin definition as follows:
+ *
+ * {@code
+ * @Override
+ *   public void define(Context context) {
+ *     if (context instanceof SonarLintContext) {
```

Comment:
I believe this code sample wouldn't work in a scanner runtime (class `SonarLintContext` won't exist) except if the class is loaded via e.g `try { class.forName(...) } ...`.

Reply:
Correct. Should I just keep `This interface is implemented by the Plugin.Context instance provided to plugins in {link {@link org.sonar.api.Plugin#define(Plugin.Context)}}` ?